utils/userFunction.js

```diff
@@ -0,0 +1,107 @@
+const currentDate = new Date();
+export function hasStudiedRecently(wordList) {
+  const currentDate = new Date();
+  if (!wordList) {
+    return;
+  }
+
+  if (wordList) {
+    const hasRecentReview = wordList.some((entry) => {
+      const reviewDate = new Date(entry.study.lastReview);
+      const comparisonDate = new Date(
+        reviewDate.setDate(reviewDate.getDate() + 1)
+      );
+
+      return comparisonDate <= currentDate;
+    });
+
+    if (hasRecentReview) {
+      return true;
+    }
+    return false;
+  }
+}
+
+function streakHasBeenSet(activeUser) {
+  const lastStreakUpdate = new Date(activeUser.lastStreakUpdate);
+  const updateDate = new Date(
+    lastStreakUpdate.setDate(lastStreakUpdate.getDate() + 1)
+  );
+
+  return updateDate >= currentDate;
+}
+
+export async function handleStreakUpdate({ activeUser, wordList }) {
```

Comment:
If it works, it works, but I'm pretty sure this could use some refactoring in the future. :)

Reply:
Yes, I will have a look :)